ui/src/app/components/configuration-panel.tsx

```diff
@@ -4,15 +4,39 @@ import * as fp from 'lodash/fp';
 
 import { BillingStatus } from 'generated/fetch';
 
+import { UIAppType } from 'app/components/apps-panel/utils';
 import { workspacesApi } from 'app/services/swagger-fetch-clients';
-import { cond, withCurrentWorkspace } from 'app/utils';
+import { cond, withCurrentWorkspace, withUserProfile } from 'app/utils';
+import { ProfileStore } from 'app/utils/stores';
+import { WorkspaceData } from 'app/utils/workspace-data';
 
 import { CromwellConfigurationPanel } from './cromwell-configuration-panel';
-import { RuntimeConfigurationPanel } from './runtime-configuration-panel';
+import {
+  RuntimeConfigurationPanel,
+  RuntimeConfigurationPanelProps,
+} from './runtime-configuration-panel';
 import { DisabledPanel } from './runtime-configuration-panel/disabled-panel';
 
-export const ConfigurationPanel = fp.flow(withCurrentWorkspace())(
-  ({ onClose, workspace, type = 'runtime', runtimeConfPanelInitialState }) => {
+export interface ConfigurationPanelProps {
+  onClose: () => void;
+  type: UIAppType;
+  runtimeConfPanelInitialState?: RuntimeConfigurationPanelProps['initialPanelContent'];
+}
+
+export const ConfigurationPanel = fp.flow(
+  withCurrentWorkspace(),
+  withUserProfile()
+)(
+  ({
+    onClose,
+    workspace,
+    type,
+    runtimeConfPanelInitialState = null,
+    profileState,
+  }: ConfigurationPanelProps & {
```

Comment:
I am unclear as to what is happening here. Is profileState needed multiple times?

Reply:
Does the commit message clarify this?

> Profile information is needed by all UI app types. Like workspaces,
I moved fetching that information slightly higher to simplify downstream
components.
>
> Note that the only change to PanelMain in this commit is the removal
of fp.flow, everything else is indent changes.

Reply:
Not quite, I don't understand the bitwise AND that you are using.
It seems like you are saying that the input to Configuration panel is the combination of ConfigurationPanelProps (which includes profileState) and {workspace: WorkspaceData;
    profileState: ProfileStore;}

Reply:
You're right about how the `&` symbol works in TypeScript. However, I don't think ConfigurationPanelProps includes `profileState`:

```
export interface ConfigurationPanelProps {
  onClose: () => void;
  type: UIAppType;
  runtimeConfPanelInitialState?: RuntimeConfigurationPanelProps['initialPanelContent'];
}
```

FWIW I think there are clearer ways to represent how the props are passed through but I haven't experimented with it yet.